### PR TITLE
 #170947544 implements signup endpoint with JWT

### PR DESCRIPTION
**Pull request by bbaime98**
#### What does this PR do?
- add successfully registration with JWT functionality
#### Description of Task to be completed?
-  add signup route
- add auth controller
- add register user function
- add search in database function
- add provide token function
- add response handler functions
#### How should this be manually tested?
- Clone repository
- run:  git checkout ft-registration-jwt-170947544
- start server: npm run dev
- use `/api/v1/auth/register`
- Enter all first and last name, email and password
#### What are the relevant pivotal tracker stories?
- [#170947544](https://www.pivotaltracker.com/story/show/170947544)
#### Screenshots (if appropriate)
- Success
image

- Fail
image

#### Questions:

### CODE REVIEW

**Review by Baraka-Mugisha — CHANGES_REQUESTED** [review]
Could you change the status code for sign up to be  201 instead of 200?

**Review by Ntare22 — CHANGES_REQUESTED** [review]
Fix the hound error and I think you'll be good to go.

**Comment by bbaime98** [comment]
it is fixed now.

File: src/tests/signupMockData.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const signupData = [
+  {
+    firstName: 'bienaime',
+    lastName: 'jeanb',
+    email: 'aime@andela.com',
```
Comment (Mnickii) [inline src/tests/signupMockData.js]: lets use the seeded data for our tests and remove this file

File: src/utils/searchInDatabase.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * @class DatabaseSearch
+ */
+class DatabaseSearch {
+/**
+ *
+ *
+ * @static
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} next
```
Comment (Mnickii) [inline src/utils/searchInDatabase.js]: Lets have this as a User controller, instead of a util.

**Review by Mnickii — CHANGES_REQUESTED** [review]

File: src/utils/searchInDatabase.js (hunk 2 above)
Comment (bbaime98) [reply src/utils/searchInDatabase.js]: sure. Am going to  update it.

File: src/index.js [hunk 3]
```diff
@@ -15,6 +16,12 @@ const port = process.env.PORT || 3000;
 
 app.use('/api', welcome);
 app.use('/api-doc', swagger);
+app.get('/', (req, res) => res.status(200).send({
```
Comment (Mnickii) [inline src/index.js]: our root route is /api

File: src/tests/inputsMockData.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const validInputs = [
+  {
+    firstName: 'bienaime',
+    lastName: 'jeanb',
```
Comment (Mnickii) [inline src/tests/inputsMockData.js]: We don't need this file if we have seeders

**Review by Mnickii — CHANGES_REQUESTED** [review]

**Comment by Mnickii** [comment]
Include testing details on PR description

**Review by Ntare22 — DISMISSED** [review]
LGTM

**Review by Baraka-Mugisha — DISMISSED** [review]

**Pull request merged**